packages/peer-store/src/utils/dedupe-addresses.ts

```diff
@@ -26,26 +26,51 @@ export async function dedupeFilterAndSortAddresses (peerId: PeerId, filter: Addr
       continue
     }
 
-    const isCertified = addr.isCertified ?? false
     const maStr = addr.multiaddr.toString()
-    const existingAddr = addressMap.get(maStr)
+    let existingAddr = addressMap.get(maStr)
 
-    if (existingAddr != null) {
-      addr.isCertified = existingAddr.isCertified || isCertified
-    } else {
-      addressMap.set(maStr, {
-        multiaddr: addr.multiaddr,
-        isCertified
-      })
+    if (existingAddr == null) {
+      existingAddr = {
+        multiaddr: addr.multiaddr
+      }
+
+      addressMap.set(maStr, existingAddr)
+    }
+
+    if (addr.isCertified === true) {
+      existingAddr.isCertified = true
+    }
+
+    if (addr.lastFailure != null) {
+      existingAddr.lastFailure = Number(addr.lastFailure)
+    }
+
+    if (addr.lastSuccess != null) {
+      existingAddr.lastSuccess = Number(addr.lastSuccess)
```

Comment:
The value is a timestamp, the max value of which can exceed 32 bits so we need to use a 64 bit number in the protobuf to store it, which means we need to represent it as a BigInt, but this is overkill for actual time values.

We can refactor this once https://github.com/ipfs/protons/issues/112 is implemented to have protons serialize/deserialize to `Number` instead of `BigInt`.